### PR TITLE
feat: Get master upto date

### PR DESCRIPTION
**Pull request by roerohan**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+mongoose.Promise = global.Promise;
 
 app.engine('hbs', exphbs({
     defaultLayout: 'main',
     extname: '.hbs'
 }));
 // Setting template Engine
 app.set('view engine', 'hbs');
-app.use('/static', express.static(__dirname + '/public'))
+app.use('/public', express.static(__dirname + '/public'))
```
Comment (roerohan) [inline index.js:21]: Recommend to use `path.join` or `path.resolve`.

File: index.js [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
 
 // parse application/x-www-form-urlencoded aka your HTML <form> tag stuff
 app.use(bodyParser.urlencoded({ extended: false }))
 
 // parse application/json aka whatever you send as a json object
 app.use(bodyParser.json())
 
-app.use('/leaderboard', leaderboardRouter);
+app.use('/api', require('./routers/api'));
```
Comment (roerohan) [inline index.js:29]: All `require`s should be on top of the page

File: models/questions.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+//create Question Schema & model
+const QuestionSchema = new Schema({
+    qno:Number,
+    ans:String
+    });
+
+const Question = mongoose.model('question',QuestionSchema);
```
Comment (roerohan) [inline models/questions.js:10]: No ` ` after `,`, this is why you use a linter.

File: routers/api.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+//show registration page
+router.get('/user/register', (_req, res) => {
+    res.render('register.hbs');
+});
+
+router.post("/user/login", async (req, res) => {
+    try {
+        const { email } = req.body;
```
Comment (roerohan) [inline routers/api.js:19]: Why not `{ email, pass }`

File: routers/api.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        const passw = req.body.pass;
+        const usr = await User.findOne({ email });
```
Comment (roerohan) [inline routers/api.js:21]: Mongoose injection spotted.

File: routers/api.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        pass = usr.pass;
+        uname = usr.uname;
```
Comment (roerohan) [inline routers/api.js:23]:
```js
const { pass, uname } = usr;
```

File: routers/api.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        res.status(500).send();
+    }
+});
+
+//add new user to the db with hashing
+router.post('/user/register', async (req, res) => {
+    //console.log(req.body);
+    const passw = req.body.pass;
+    const { email, uname, regno } = req.body;
+    if(req.body.pass.length<25 && req.body.email.length<100 && req.body.uname.length<25 && (req.body.regno.length==9 || req.body.regno.length==0)){
```
Comment (roerohan) [inline routers/api.js:39]: I mean, you already stored `req.body.email` etc. in a variable, just use that again.
Also, mongoose injection again.

File: routers/leaderboardApi.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+// 		.catch(err => console.log(err));
+
+// 	res.status(200).json({
+// 		message: "Question object created",
+// 		question: question
+// 	});
+// });
+
+//-----------------------------------------------------
+
```
Comment (roerohan) [inline routers/leaderboardApi.js:50]: Random comments discouraged unless they're gonna be used later

File: routers/leaderboardApi.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+		console.log(e)
+		res.status(500).json({
+			error: e
+		});
+	}
+});
+
+// Get information on a specific user
+router.get('/user', async (req, res, next) => {
+	const uname = req.body.uname?.toString();
```
Comment (roerohan) [inline routers/leaderboardApi.js:71]: Finally prevented mongoose injection in 1 route.

File: routers/leaderboardApi.js [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+	if (!uname){
+		res.status(500).json({
+			error: "Invalid input given",
+			code: 1
+		});
+		return;
+	}
+
+	const userData = await User.findOne({ uname: uname });
```
Comment (roerohan) [inline routers/leaderboardApi.js:80]: Same as `User.findOne({ uname })`

File: routers/leaderboardApi.js [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+			error: "Input too large",
+			code: 2,
+			result: false
+		});
+	}
+
+	console.log(qno, ans, uname, pass);
+
+	const userData = await User.findOne({
+		uname: uname
```
Comment (roerohan) [inline routers/leaderboardApi.js:128]: ESLint recommended

File: routers/leaderboardApi.js [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+	if (!passCheck) {
+		return res.status(500).json({
+			error: "Password incorrect",
+			code: 4,
+			result: false
+		});
+	}
+
+	const qa = await Question.findOne({
+		qno: qno,
```
Comment (roerohan) [inline routers/leaderboardApi.js:152]: Are you sure `qno` will work if it's a string?

**Review by roerohan — COMMENTED** [review]
Well, check this

File: routers/leaderboardApi.js (hunk 12 above)
Comment (UnknownAbyss) [reply routers/leaderboardApi.js:152]: Yes that has been tested

File: routers/leaderboardApi.js (hunk 11 above)
Comment (UnknownAbyss) [reply routers/leaderboardApi.js:128]: Will do :p

File: routers/leaderboardApi.js (hunk 9 above)
Comment (UnknownAbyss) [reply routers/leaderboardApi.js:71]: Its there in all routes right?

File: index.js (hunk 1 above)
Comment (UnknownAbyss) [reply index.js]: Solved

**Pull request merged**